src/api/routes/core-node-rpc-proxy.ts

```diff
@@ -121,6 +153,9 @@ export const CoreNodeRpcProxyRouter: FastifyPluginAsync<
     upstream: `http://${stacksNodeRpcEndpoint}`,
     rewritePrefix: '/v2',
     http2: false,
+    // Use the `node:http` agent for requests https://github.com/fastify/fastify-reply-from?tab=readme-ov-file#globalagent
+    globalAgent: true,
+    http: {},
```

Comment:
Fastify defaults to using Fetch/undici which isn't supported by the `nock` testing lib used to mock network responses. These settings cause it to use the `node:http` global agent. From what I understand, it's a bit slower but probably fine.

Reply:
Curious, have you tried using the `MockAgent`? https://github.com/hirosystems/token-metadata-api/blob/0c02a159290925963d516403d712582e79c7db2d/tests/token-queue/process-smart-contract-job.test.ts#L44-L56

Reply:
Nice looks like that works https://github.com/hirosystems/stacks-blockchain-api/pull/2172/commits/6ab509e5087f9601ef3ba52479a3cd66b3000490